modin/pandas/utils.py

```diff
@@ -105,3 +107,35 @@ def is_scalar(obj):
     from .base import BasePandasDataset
 
     return not isinstance(obj, BasePandasDataset) and pandas_is_scalar(obj)
+
+
+def from_modin_frame(df, sortorder=None, names=None):
+    """
+    Make a MultiIndex from a DataFrame.
+
+    Parameters
+    ----------
+    df : DataFrame
+        DataFrame to be converted to MultiIndex.
+    sortorder : int, optional
+        Level of sortedness (must be lexicographically sorted by that
+        level).
+    names : list-like, optional
+        If no names are provided, use the column names, or tuple of column
+        names if the columns is a MultiIndex. If a sequence, overwrite
+        names with the given sequence.
+
+    Returns
+    -------
+    MultiIndex
+        The MultiIndex representation of the given DataFrame.
+    """
+    from .dataframe import DataFrame
+
+    if isinstance(df, DataFrame):
+        df = df._to_pandas()
+    return _original_pandas_MultiIndex_from_frame(df, sortorder, names)
```

Comment:
What is the need to use this additional function?

Reply:
What do you mean "additional function"?

Reply:
I mean using this line `MultiIndex.from_frame(df, sortorder, names)`. I think the line 143 is extra.

Reply:
That is how it I initially implemented it, but it results in an infinite recursion because `MultiIndex.from_frame` is `from_modin_frame_to_mi` now. This is a patch which makes a change that you request:

```patch
diff --git a/modin/pandas/utils.py b/modin/pandas/utils.py
index e151d7c..4c06005 100644
--- a/modin/pandas/utils.py
+++ b/modin/pandas/utils.py
@@ -137,8 +137,7 @@ def from_modin_frame_to_mi(df, sortorder=None, names=None):

         ErrorMessage.default_to_pandas("`MultiIndex.from_frame`")
         df = df._to_pandas()
-    return _original_pandas_MultiIndex_from_frame(df, sortorder, names)
+    return MultiIndex.from_frame(df, sortorder, names)


-_original_pandas_MultiIndex_from_frame = MultiIndex.from_frame
 MultiIndex.from_frame = from_modin_frame_to_mi
```
and this is how it works:
```
In [2]: import modin.pandas as pd
UserWarning: Modin Ray engine was started with 15 GB free space avaliable, if it is not enough for your application, please set environment variable MODIN_ON_RAY_PLASMA_DIR=/directory/without/space/limiting

In [3]: import pandas

In [4]:     mdf = pd.DataFrame(
   ...:         [["bar", 1, "1", 1.1], ["bar", 2, "2", 2.2], ["foo", 1, "3", 3.3], ["foo", 2, "4", 4.4]],
   ...:             columns=["first", "second", "data_str", "data_f"],
   ...:             )
UserWarning: Distributing <class 'list'> object. This may take some time.

In [5]:     pdf = pandas.DataFrame(
   ...:         [["bar", 1, "1", 1.1], ["bar", 2, "2", 2.2], ["foo", 1, "3", 3.3], ["foo", 2, "4", 4.4]],
   ...:             columns=["first", "second", "data_str", "data_f"],
   ...:             )

In [6]: pd.MultiIndex.from_frame(pdf)
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
<ipython-input-6-01e1da826c3c> in <module>
----> 1 pd.MultiIndex.from_frame(pdf)

/localdisk/gashiman/modin/modin/pandas/utils.py in from_modin_frame_to_mi(df, sortorder, names)
    138         ErrorMessage.default_to_pandas("`MultiIndex.from_frame`")
    139         df = df._to_pandas()
--> 140     return MultiIndex.from_frame(df, sortorder, names)
    141
    142

... last 1 frames repeated, from the frame below ...

/localdisk/gashiman/modin/modin/pandas/utils.py in from_modin_frame_to_mi(df, sortorder, names)
    138         ErrorMessage.default_to_pandas("`MultiIndex.from_frame`")
    139         df = df._to_pandas()
--> 140     return MultiIndex.from_frame(df, sortorder, names)
    141
    142

RecursionError: maximum recursion depth exceeded while calling a Python object

In [7]: pd.MultiIndex.from_frame(mdf)
UserWarning: `MultiIndex.from_frame` defaulting to pandas implementation.
To request implementation, send an email to feature_requests@modin.org.
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
<ipython-input-7-31daf7b77262> in <module>
----> 1 pd.MultiIndex.from_frame(mdf)

/localdisk/gashiman/modin/modin/pandas/utils.py in from_modin_frame_to_mi(df, sortorder, names)
    138         ErrorMessage.default_to_pandas("`MultiIndex.from_frame`")
    139         df = df._to_pandas()
--> 140     return MultiIndex.from_frame(df, sortorder, names)
    141
    142

... last 1 frames repeated, from the frame below ...

/localdisk/gashiman/modin/modin/pandas/utils.py in from_modin_frame_to_mi(df, sortorder, names)
    138         ErrorMessage.default_to_pandas("`MultiIndex.from_frame`")
    139         df = df._to_pandas()
--> 140     return MultiIndex.from_frame(df, sortorder, names)
    141
    142

RecursionError: maximum recursion depth exceeded while calling a Python object
```

Reply:
Oh, I see.